e2e/e2e_tests.py

```diff
@@ -0,0 +1,151 @@
+import os
+from time import sleep
+import unittest
+import subprocess
+
+try:
+    import requests
+    from bs4 import BeautifulSoup
+except ImportError:
+    requirements_file = os.path.join(os.path.dirname(__file__), 'requirements.txt')
+    subprocess.check_call("pip install -r {}".format(requirements_file).split(' '))
```

Comment:
A test that installs its own dependencies? Shouldn't it rather fail with a hint on how the user _can_ install them, if they choose so?

As it is itself written in python, doesn't `pip` come with a python API, so we don't have to make a system call to invoke it?